lib/shared/config-manager/src/index.ts

```diff
@@ -63,26 +84,115 @@ export class EnvironmentConfigManager {
                 this.logger.debug('DevCycle initial config loaded')
             })
             .finally(() => {
-                if (this.disablePolling) {
-                    return
-                }
-                this.intervalTimeout = this.setInterval(async () => {
-                    try {
-                        await this._fetchConfig()
-                    } catch (ex) {
-                        this.logger.error((ex as Error).message)
-                    }
-                }, this.pollingIntervalMS)
+                this.startPolling(this.configPollingIntervalMS)
+                this.startSSE()
             })
     }
 
-    stopPolling(): void {
-        this.disablePolling = true
+    private startSSE(): void {
+        if (!this.enableRealtimeUpdates) return
+
+        if (!this.configSSE) {
+            this.logger.warn('No SSE configuration found')
+            return
+        }
+        if (this.sseConnection) {
+            return
+        }
+
+        const url = new URL(
+            this.configSSE.path,
+            this.configSSE.hostname,
+        ).toString()
+        this.logger.debug(`Starting SSE connection to ${url}`)
+
+        this.sseConnection = new SSEConnection(url, this.logger, {
+            onMessage: this.onSSEMessage.bind(this),
+            onOpen: () => {
+                this.logger.debug('SSE connection opened')
+                // Set config polling interval to 10 minutes
+                this.startPolling(this.sseConfigPollingIntervalMS)
+            },
+            onConnectionError: () => {
+                this.logger.debug('SSE connection error, switching to polling')
+                // reset polling interval to default
+                this.startPolling(this.configPollingIntervalMS)
+                this.stopSSE()
+            },
```

Comment:
Is there a risk that the polling does not start?

Reply:
nope